src/Lecture3.hs

```diff
@@ -181,7 +186,12 @@ together only different elements.
 Product {getProduct = 6}
 
 -}
-appendDiff3 = error "TODO"
+appendDiff3 :: (Eq a) => [a] -> [a] -> [a] -> [a]
```

Comment:
Your current implementation works only for lists. You can generalize it to any `Semigroup` and use the `<>` operator in the implementation to make the function more polymorphic 💠 
```suggestion
appendDiff3 :: (Eq a, Semigroup a) => a -> a -> a -> a
```